### PR TITLE
GGT-2034 - New provider for react library

### PR DESCRIPTION
**Pull request by alanko0511**
Problems:
- No proper error if the user doesn't include the provider.
- urql will create a client that points to `/graphql` if the provider is not included.
- urql has no parameter to get the url of the client.

Solution:
- Create a provider that wrap the urql provider, then check to see if that provider's context has a client in it.

### CODE REVIEW

**Review by thegedge — COMMENTED** [review]
Nice! I think we are able to and should write a test for this one @alanko0511 :)

File: packages/react/src/useStructuralMemo.ts
Comment (alanko0511) [reply packages/react/src/useStructuralMemo.ts:10]: I think the second option would be better. That way when we create a new hook in the future, we wouldn't forget to add this context checking code.

Comment (thegedge) [reply packages/react/src/useStructuralMemo.ts:10]: I'm down. Let's make that happen!

I think this should be easy to test, but lemme know if there's any difficulty setting this up for testing. Happy to pair on it :)

Comment (LucasUnplugged) [reply packages/react/src/useStructuralMemo.ts:10]: Aside from this point, the PR looks good. But I'll wait until the refactor to approve, after re-reviewing.

**Review by LucasUnplugged — APPROVED** [review]
Nice! LGTM!

**Review by thegedge — APPROVED** [review]
Oops, forgot to send off my ✅

**Pull request merged**
